cristin-project/src/test/java/no/unit/nva/cristin/projects/update/UpdateCristinProjectHandlerTest.java

```diff
@@ -127,20 +136,60 @@ void shouldReturnBadRequestWhenTitleHasNoLanguageFieldPresent() throws IOExcepti
     }
 
     @Test
-    void shouldReturnOKNoContentWhenRequestHasTitleAndLanguagePresent() throws IOException {
-        var jsonObject = OBJECT_MAPPER.createObjectNode();
-        jsonObject.put(TITLE, randomString());
-        jsonObject.put(LANGUAGE, toUri(randomLanguage()).toString());
-        GatewayResponse<Void> gatewayResponse = sendQuery(jsonObject.toString());
-        assertEquals(HTTP_NO_CONTENT, gatewayResponse.getStatusCode());
+    void shouldReturnBadRequestWhenFundingIsMissingRequiredFieldFundingSourceCode() throws IOException {
+        var input = minimalJsonProject();
+        var fundingWithoutRequiredFieldCode = new Funding(new FundingSource(randomNamesMap(), null), randomString());
+        input.putPOJO(FUNDING, List.of(fundingWithoutRequiredFieldCode));
+        GatewayResponse<Void> gatewayResponse = sendQuery(input.toPrettyString());
+
+        assertThat(gatewayResponse.getBody(), containsString(FUNDING_MISSING_REQUIRED_FIELDS));
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenFundingIsNotAnArray() throws IOException {
+        var input = minimalJsonProject();
+        var notAList = randomString();
+        input.put(FUNDING, notAList);
+        GatewayResponse<Void> gatewayResponse = sendQuery(input.toPrettyString());
+
+        assertThat(gatewayResponse.getBody(), containsString(format(MUST_BE_A_LIST, FUNDING)));
+    }
```

Comment:
I would be tempted to create a "invalidFundingProvider" to test the cases in a parameterized way

Reply:
Can we then pass the exception message as well as a parameterized exception?

Reply:
Think i found a solution. will update